### PR TITLE
Editor Module: Replace connect usage with withSelect/withDispatch (1st round)

### PR DESCRIPTION
**Pull request by youknowriad**
Just getting rid of some legacy. and to avoid huge PRs, I'm updating 20 components in this first round. There's 45 remaining usage of `connect` in the editor module.

**Testing instructions**

Check that:
 
 - autosaves triggers
 - the block movers work
 - the block toolbar work
 - the block settings menu work
 - the in between inserter works
 - the default appender works.
 - the block inspector works
 - multiselection works
 - transformations work.

I already tested everything, so it should be ok. I'm willing to merge this pretty quickly to avoid a lot of conflicts.

### CODE REVIEW

File: editor/components/block-switcher/index.js [hunk 1]
```diff
@@ -102,21 +96,19 @@ export function BlockSwitcher( { blocks, onTransform, isLocked } ) {
...
 
 export default compose(
-	connect(
-		( state, ownProps ) => {
-			return {
-				blocks: ownProps.uids.map( ( uid ) => getBlock( state, uid ) ),
-			};
+	withSelect( ( select, ownProps ) => {
+		return {
+			blocks: ownProps.uids.map( ( uid ) => select( 'core/editor' ).getBlock( uid ) ),
```
Comment (gziolo) [inline editor/components/block-switcher/index.js]: Not related to your change, but `map` usage in here looks suspicious. It probably rerenders too often.

File: editor/store/actions.js [hunk 2]
```diff
@@ -208,6 +208,28 @@ export function replaceBlock( uid, block ) {
...
+	return ( uids, rootUID ) => {
+		return {
+			type,
+			uids,
+			rootUID,
+		};
+	};
+}
+
+export const moveBlocksDown = createOnMove( 'MOVE_BLOCKS_DOWN' );
```
Comment (gziolo) [inline editor/store/actions.js]: Partial application ❤️

**Review by gziolo — COMMENTED** [review]
I left my comments, it all looks great. Happy to see how it looks after all changes get applied. In my opinion `data` module plays nicely! 

I haven't found time to test. I can do it tomorrow.

File: editor/components/block-switcher/index.js (hunk 1 above)
Comment (gziolo) [reply editor/components/block-switcher/index.js]: Now I disagree with myself, as long as it is the only prop passed, it is fine :)

**Review by gziolo — APPROVED** [review]
Nice work, I haven't found any regressions. Let's ship it :shipit:

File: editor/components/block-list/block-html.js [hunk 3]
```diff
@@ -59,13 +54,13 @@ class BlockHTML extends Component {
 	}
 }
 
-export default connect(
-	( state, ownProps ) => ( {
-		block: getBlock( state, ownProps.uid ),
-	} ),
-	{
+export default compose( [
```
[the remark this answers is not in the document]
Comment (aduth) [reply editor/components/block-list/block-html.js]: >@aduth, should we add Eslint rule to enforce arrays for `compose`? I always forgot to add it :)

Meh. 

"Developer preference" 😃

**Pull request merged**